tket/src/Mapping/include/Mapping/LexiRoute.hpp

```diff
@@ -43,35 +43,59 @@ class LexiRoute {
    */
   LexiRoute(
       const ArchitecturePtr& _architecture,
-      std::shared_ptr<MappingFrontier>& _mapping_frontier);
+      MappingFrontier_ptr& _mapping_frontier);
 
   /**
    * When called, LexiRoute::solve will modify the Circuit held in
    * MappingFrontier object passed at class construction. Either a SWAP gate
    * will be inserted at the input boundary of the held Circuit or a CX gate
    * will be transformed into a BRIDGE gate. The added SWAP or BRIDGE gate will
-   * be valid for the Architecture passed at class construction. Additionally,
-   * an "unlabelled" Qubit in the Circuit may be relabelled to a Node in the
-   * Architecture, or an "unlabelled" Qubit may have its path merged with an
-   * ancilla qubit.
+   * be valid for the Architecture passed at class construction.
    * The decision making is based on the heuristic outlined in arXiv:1902.08091.
    *
    * @param lookahead Number of slices to lookahead at when determining best
    * SWAP or BRIDGE
+   *
+   * @return True if solve has modified circuit for mapping purposes
    */
-  void solve(unsigned lookahead);
+  bool solve(unsigned lookahead);
+
+  /**
+   * When called an "unlabelled" Qubit in the Circuit may be relabelled to a
+   * Node in the Architecture, or an "unlabelled" Qubit may have its path merged
+   * with an ancilla qubit. The decision making is based on the heuristic
+   * outlined in arXiv:1902.08091.
+   *
+   * @return True if solve_labelling has modified circuit for mapping purposes
+   */
+  bool solve_labelling();
 
  private:
+  /** Only considers two-qubit vertices if both qubits are labelled to
+   * Architecture */
+  enum class AssignedOnly { Yes, No };
+  /** Returns a bool confirming if vertices are valid for LexiRoute::solve */
+  enum class CheckRoutingValidity { Yes, No };
+  /** Returns a bool confirming if vertices are valid for
+   * LexiRoute::solve_labelling */
+  enum class CheckLabellingValidity { Yes, No };
+
   /**
    * this->interacting_uids_ attribute is a map where key is one UnitID
    * and value is the UnitID it needs to be adjacent to.
    * This map is implicitly updated whenever a logical SWAP is inserted.
    * set_interacting_uids determines this map for the first parallel set of
    * interacting UnitID in the Circuit held in this->mapping_frontier_
-   * @param assigned_only If true, only include interactions where both UnitID
+   * @param assigned_only If Yes, only include interactions where both UnitID
    * are in this->architecture_.
+   * @param route_check If Yes, return false if solve not possible
+   * @param label_check If Yes, return false if solve_labelling not possible
+   *
+   * @return bool depending on ENUM conditions
    */
-  void set_interacting_uids(bool assigned_only = false);
+  bool set_interacting_uids(
+      AssignedOnly assigned_only, CheckRoutingValidity route_check,
```

Comment:
Was it intentional to remove the default values?

Reply:
Yeah, I decided it probably obfuscated things further when the use is somewhat varied